src/Engine/ProtoCore/RuntimeStatus.cs

```diff
@@ -70,7 +71,16 @@ public IOutputStream WebMessageHandler
             set;
         }
 
+        [Obsolete("Property will be deprecated in Dynamo 3.0. Please use WarningEntries instead")]
         public IEnumerable<Runtime.WarningEntry> Warnings
+        {
+            get
+            {
+                return warnings.ToList();
```

Comment:
I'm confused, this had no setter, and this getter copies the list of Warnings - where did you find concurrent access to the underlying collection where the existing getter caused a problem?

Reply:
I managed to cause a crash a couple of times ...but not reliably. The Warnings List was being modified on one thread while it was also being iterated on another thread. 
Happened while moving MouseManipulators around...